numerics/poisson_series_basis_body.hpp

```diff
@@ -16,125 +16,201 @@ using geometry::Hilbert;
+  static Coefficient Unit();
+};
+
+// Specialization for quantities.
+template<typename Coefficient, std::size_t... ds>
+struct CoefficientGenerator<Coefficient,
+                            /*dimension=*/1,
+                            std::index_sequence<ds...>,
+                            std::enable_if_t<is_quantity_v<Coefficient>>> {
+  // Returns a unit quantity.  This function is templated for consistency with
+  // the preceeding specialization; |d| must be 0.
+  template<int d>
+  static Coefficient Unit();
 };
 
-template<typename Series, int n, int d>
-Series SeriesGenerator<Series, n, d>::Aperiodic(Instant const& origin) {
-  return Series(Unit<typename Series::AperiodicPolynomial>(origin), {});
+template<typename Coefficient, int dimension, std::size_t... ds>
+template<int d>
+Coefficient
+CoefficientGenerator<Coefficient,
+                     dimension,
+                     std::index_sequence<ds...>,
+                     std::enable_if_t<!is_quantity_v<Coefficient>>>::Unit() {
+  using Scalar = typename Hilbert<Coefficient>::NormType;
+  return Coefficient({(d == ds ? si::Unit<Scalar> : Scalar{})...});
 }
 
-template<typename Series, int n, int d>
-Series SeriesGenerator<Series, n, d>::Sin(AngularFrequency const& ω,
-                                          Instant const& origin) {
-  typename Series::AperiodicPolynomial const aperiodic_zero{{}, origin};
-  typename Series::PeriodicPolynomial const periodic_zero{{}, origin};
-  return Series(aperiodic_zero,
-                {{ω,
-                  {/*sin=*/Unit<typename Series::PeriodicPolynomial>(origin),
-                   /*cos=*/periodic_zero}}});
+template<typename Coefficient, std::size_t... ds>
+template<int d>
+Coefficient
+CoefficientGenerator<Coefficient,
+                     /*dimension=*/1,
+                     std::index_sequence<ds...>,
+                     std::enable_if_t<is_quantity_v<Coefficient>>>::Unit() {
+  static_assert(d == 0);
+  return si::Unit<Coefficient>;
 }
 
-template<typename Series, int n, int d>
-Series SeriesGenerator<Series, n, d>::Cos(AngularFrequency const& ω,
-                                          Instant const& origin) {
-  typename Series::AperiodicPolynomial const aperiodic_zero{{}, origin};
-  typename Series::PeriodicPolynomial const periodic_zero{{}, origin};
-  return Series(
-      aperiodic_zero,
-      {{ω,
-        {/*sin=*/periodic_zero,
-         /*cos=*/Unit<typename Series::PeriodicPolynomial>(origin)}}});
+
+// In the following classes |index| and |indices...| encode the dimension, the
+// degree and the parity (Sin or Cos) of a series.  How this encoding is
+// performed determines the order of the series in the basis.
+
+// A helper to build unit polynomials.  |Polynomial| must be a polynomial with
+// values in a Hilbert space with |dimension| dimensions.
+template<typename Polynomial, int dimension>
+struct PolynomialGenerator {
+  // Returns a polynomial whose dimension and degree are encoded in |index|.
```

Comment:
s/dimension/coordinate/, using "dimension" in both the "nth dimension" and "number of dimensions" sense is confusing.

Same everywhere else (including in the comments).